src/WidgetFactory.php

```diff
@@ -109,7 +112,19 @@ public static function createWidget(array $config, ?string $theme = null): Widge
             }
         }
 
-        return self::$factory->create($config);
+        try {
+            return self::$factory->create($config);
+        } catch (NotInstantiableException $exception) {
+            if (self::$initialized) {
+                throw $exception;
+            }
+
+            /**
+             * @var string $className When `$className` is not string `$factory->create()`
+             * throws not `NotInstantiableException` exception.
```

Comment:
What does it mean?

Reply:
```php
$className = $config[ArrayDefinition::CLASS_NAME] ?? null;
```

Here `$className` is always string, because if not, then factory throws `InvalidArgumentException`, not `NotInstantiableException`.